source/slang/slang-check-expr.cpp

```diff
@@ -1365,107 +1365,97 @@ return coerce(typeExp.type, initListExpr);
         swizExpr->loc = memberRefExpr->loc;
         swizExpr->base = memberRefExpr->baseExpression;
 
-        int elementIndices[4][2];
+        // We can have up to 4 swizzles of two elements each
+        MatrixCoord elementIndices[4];
         int elementCount = 0;
 
         bool anyDuplicates = false;
-        int oneIndexed = -1;
+        int zeroIndexOffset = -1;
 
         String swizzleText = getText(memberRefExpr->name);
-        auto coursor = swizzleText.begin();
+        auto cursor = swizzleText.begin();
 
-        while (coursor != swizzleText.end())
+        // The contents of the string are 0-terminated
+        // Every update to cursor corresponds to a check against 0-termination
+        while (cursor != swizzleText.end())
         {
             // Throw out swizzling with more than 4 output elements
             if (elementCount >= 4) {
                 getSink()->diagnose(swizExpr, Diagnostics::invalidSwizzleExpr, swizzleText, baseElementType->toString());
                 return CreateErrorExpr(memberRefExpr);
             }
-            int elementIndex[2];
-            elementIndex[0] = 0;
-            elementIndex[1] = 0;
+            MatrixCoord elementIndex = { 0, 0 };
```

Comment:
Nit pick: I guess it could be argued that the MatrixCoord is an 'index', but when I read index it means an integer based value. That matrixCoord, or just coord avoids this.